File: code-review-graph/tests/fixtures/sample.kt
package com.example

import java.util.UUID

interface UserRepository {
    fun findById(id: Int): User?
    fun save(user: User)
}

data class User(val id: Int, val name: String, val email: String)

class InMemoryRepo : UserRepository {
    private val users = mutableMapOf<Int, User>()

    override fun findById(id: Int): User? = users[id]

    override fun save(user: User) {
        users[user.id] = user
        println("Saved user ${user.id}")
    }
}

fun createUser(repo: UserRepository, name: String, email: String): User {
    val user = User(1, name, email)
    repo.save(user)
    return user
}

File: code-review-graph/tests/fixtures/sample.swift
import Foundation

protocol UserRepository {
    func findById(_ id: Int) -> User?
    func save(_ user: User)
}

struct User {
    let id: Int
    let name: String
    let email: String
}

class InMemoryRepo: UserRepository {
    private var users: [Int: User] = [:]

    func findById(_ id: Int) -> User? {
        return users[id]
    }

    func save(_ user: User) {
        users[user.id] = user
        print("Saved user \(user.id)")
    }
}

func createUser(repo: UserRepository, name: String, email: String) -> User {
    let user = User(id: 1, name: name, email: email)
    repo.save(user)
    return user
}
